tests/unit/s2n_release_non_empty_buffers_test.c

```diff
@@ -55,6 +55,7 @@ int mock_client(struct s2n_test_io_pair *io_pair)
         _exit(2);
     }
 
+    EXPECT_SUCCESS(s2n_io_pair_close_one_end(io_pair, S2N_CLIENT));
     s2n_shutdown(conn, &blocked);
```

Comment:
Wait, this doesn't break the test? s2n_shutdown potentially both reads and writes. Is s2n_shutdown now failing, but this test doesn't have sufficient assertions to catch that?

Reply:
Yes, this change is problematic with the test. I will investigate and request review once everything is passed.

Reply:
I have moved the `s2n_io_pair_close_one_end` to be right above `s2n_cleanup()`, and tests can pass.